lib/core/base/virtual-node.js

```diff
@@ -1,15 +1,39 @@
 const whitespaceRegex = /[\t\r\n\f]/g;
 
+class AbstractVirtualNode {
+	constructor() {}
```

Comment:
Can remove this.

Reply:
I thought so too but something in our babel compiler breaks the `this` reference in the class when being extended and throws `TypeError: Cannot set property 'shadowId' of undefined`. It seems it assumes the parent class will have a `constructor` and that you need to call `super` on it to get the the correct `this` reference.

Reply:
The subclass will need to call `super()` before setting instance properties, but the superclass doesn't need a constructor in order for that to happen.

Contrived example:

```js
class AbstractVNode {
  foo() {
    throw new Error('not implemented');
  }
}

class MyVNode extends AbstractVNode {
  constructor() {
    super();
    this.banana = true;
  }
  foo() {
    return 'hi';
  }
}

const n = new MyVNode()
n.foo()
```